core/spatial_map.cpp

```diff
@@ -788,395 +788,382 @@ void SpatialMap::ColorForValue(double p_value, float *p_rgb_ptr)
 		for (int64_t b = 0; b < dim_b; ++b)
 		{
 			double coverage_b = (!periodic_b_ && ((b == 0) || (b == dim_b - 1))) ? 0.5 : 1.0;
-			
+
 			for (int64_t a = 0; a < dim_a; ++a)
 			{
 				double coverage_a = (!periodic_a_ && ((a == 0) || (a == dim_a - 1))) ? 0.5 : 1.0;
-				double coverage = coverage_a * coverage_b;					// handles partial coverage at the edges of the spatial map
-				
-				// calculate the kernel's effect at point (a,b)
+				double coverage = coverage_a * coverage_b;
+
 				double kernel_total = 0.0;
 				double conv_total = 0.0;
-				
-				for (int64_t kernel_a = 0; kernel_a < kernel_dim_a; kernel_a++)
+
+				for (int64_t kernel_b = 0; kernel_b < kernel_dim_b; kernel_b++)
 				{
-					int64_t conv_a = a + kernel_a + kernel_a_offset;
-					
-					// handle bounds: either clip or wrap
-					if ((conv_a < 0) || (conv_a >= dim_a))
+					int64_t conv_b = b + kernel_b + kernel_b_offset;
+
+					if ((conv_b < 0) || (conv_b >= dim_b))
 					{
-						if (!periodic_a_)
+						if (!periodic_b_)
 							continue;
-						
-						// periodicity: assume the two edges have identical values, skip over the edge value on the opposite side
-						while (conv_a < 0)
-							conv_a += (dim_a - 1);	// move -1 to dim - 2
-						while (conv_a >= dim_a)
-							conv_a -= (dim_a - 1);	// move dim to 1
+
+						while (conv_b < 0)
+							conv_b += (dim_b - 1);
+						while (conv_b >= dim_b)
+							conv_b -= (dim_b - 1);
 					}
-					
-					for (int64_t kernel_b = 0; kernel_b < kernel_dim_b; kernel_b++)
+
+					for (int64_t kernel_a = 0; kernel_a < kernel_dim_a; kernel_a++)
 					{
-						int64_t conv_b = b + kernel_b + kernel_b_offset;
-						
-						// handle bounds: either clip or wrap
-						if ((conv_b < 0) || (conv_b >= dim_b))
+						int64_t conv_a = a + kernel_a + kernel_a_offset;
+
+						if ((conv_a < 0) || (conv_a >= dim_a))
 						{
-							if (!periodic_b_)
+							if (!periodic_a_)
 								continue;
-							
-							// periodicity: assume the two edges have identical values, skip over the edge value on the opposite side
-							while (conv_b < 0)
-								conv_b += (dim_b - 1);	// move -1 to dim - 2
-							while (conv_b >= dim_b)
-								conv_b -= (dim_b - 1);	// move dim to 1
+
+							while (conv_a < 0)
+								conv_a += (dim_a - 1);
+							while (conv_a >= dim_a)
+								conv_a -= (dim_a - 1);
 						}
-						
-						// this point is within bounds; add it in to the totals
+
 						double kernel_value = kernel_values[kernel_a + kernel_b * kernel_dim_a] * coverage;
 						double pixel_value = values_[conv_a + conv_b * dim_a];
-						
-						// we keep a total of the kernel values that were within bounds, for this point
+
 						kernel_total += kernel_value;
-						
-						// and we keep a total of the convolution - kernel values times pixel values
 						conv_total += kernel_value * pixel_value;
 					}
 				}
-				
+
 				*(new_values_ptr++) = ((kernel_total > 0) ? (conv_total / kernel_total) : 0);
 			}
 		}
 	}
-	
-#if 0
```

Comment:
Hmm, why remove this `#if` 0 code?  It performs a useful self-check that should still work, right?  (Maybe it would need to compare with a tolerance now?  You don't need to fix it to work correctly; maybe just put a comment in it saying "needs to check with tolerance now due to SIMD"?